### PR TITLE
feat: :sparkles: ClearML support for TensorFlow

### PR DESCRIPTION
**Pull request by odulcy-mindee**
This PR introduces [ClearML](https://clear.ml/pricing/) support in references script, for TensorFlow only (PyTorch will be done in another PR).

At the moment, ClearML captures **everything**, meaning that logs, model checkpoints, etc. are tracked. Do we want to keep it like that or do we want to block everything [auto_connect_frameworks](https://clear.ml/docs/latest/docs/references/sdk/task/#taskinit) in `Task.init` by default ? If so, how do we expose parameters to users to tune this behavior ?

### CODE REVIEW

File: references/detection/train_tensorflow.py [hunk 1]
```diff
@@ -359,9 +374,9 @@ def parse_args():
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
     )
 
-    parser.add_argument("train_path", type=str, help="path to training data folder")
-    parser.add_argument("val_path", type=str, help="path to validation data folder")
     parser.add_argument("arch", type=str, help="text-detection model to train")
+    parser.add_argument("--train_path", type=str, help="path to training data folder")
+    parser.add_argument("--val_path", type=str, help="path to validation data folder")
```
Comment (odulcy-mindee) [inline references/detection/train_tensorflow.py:379]: I changed that so it's the same options than `recognition`. I'll do the same for PyTorch in another PR.

**Review by felixdittrich92 — DISMISSED** [review]
Thanks a lot @odulcy-mindee 🙏
We should keep the default config which WE use already for wandb. For specific use cases the scripts are open to be modified by users :)

**Comment by felixdittrich92** [comment]
@odulcy-mindee three small things:

For detection it should be `train_path` instead of `--train_path` because it's required we have no other option like the generators for classicication and recognition :)

Point 2: 
Can we import wandb also after the condition ?

Point 3: 
weight decay should be configurable 

:)

File: references/detection/train_tensorflow.py (hunk 1 above)
Comment (felixdittrich92) [reply references/detection/train_tensorflow.py:379]: See comment :)

**Comment by odulcy-mindee** [comment]
@felixdittrich92  
> Point 1: For detection it should be `train_path` instead of `--train_path` because it's required we have no other option like the generators for classicication and recognition :)
Oh yeah, I just wanted to have same way to provide the dataset. I can revert or make it mandatory with `¶equired=True`

> Point 2: Can we import wandb also after the condition ?
Sure !

> Point 3: weight decay should be configurable (or removed from the config If unused) ?
Ok !

**Comment by felixdittrich92** [comment]
@odulcy-mindee 
I would be fine with `required=True` 👍

**Comment by odulcy-mindee** [comment]
@felixdittrich92 I removed `weight-decay` (it was unused) and added name to the scheduler, tell me what you think about it

**Review by felixdittrich92 — APPROVED** [review]
Thanks LGTM now :)
Only one question does clear ml also ask for login like wandb ?

**Comment by felixdittrich92** [comment]
Otherwise we should add a short reference but we can do this in #1256

**Comment by odulcy-mindee** [comment]
@felixdittrich92 it looks for a `clearml.conf` in home. If the file does not exist, clearml raises an error and provides a link to doc to do the setup :+1:
